week1/1.5/Notes/Notes/Controllers/View Controllers/NoteTableViewController.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  NoteTableViewController.swift
+//  Notes
+//
+//  Created by Eric Lanza on 7/6/18.
+//  Copyright © 2018 ETLanza. All rights reserved.
+//
+
+import UIKit
+
+class NoteTableViewController: UITableViewController {
+    
+    //MARK: - IBOutlets
+    @IBOutlet weak var editBarButton: UIBarButtonItem!
+    
+    //MARK: - IBActions
+    @IBAction func editButtonPressed(_ sender: UIBarButtonItem) {
+        if  tableView.isEditing {
+            tableView.setEditing(false, animated: true);
+            editBarButton.title = "Edit";
+            editBarButton.style = UIBarButtonItemStyle.plain;
+        } else {
+            tableView.setEditing(true, animated: true);
+            editBarButton.style = UIBarButtonItemStyle.done
+            editBarButton.title = "Done";
+            editBarButton.style =  UIBarButtonItemStyle.done;
+        }
+    }
+    
+    //MARK: - Properties
+    let searchController = UISearchController(searchResultsController: nil)
+    
+    var filteredNotes = [Note]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        NoteController.shared.loadFromPersistentStore()
+        searchController.searchResultsUpdater = self
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.searchBar.placeholder = "Search notes"
+        navigationItem.searchController = searchController
+        navigationItem.hidesSearchBarWhenScrolling = false
+        definesPresentationContext = true
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+    }
+    
+    // MARK: - Table view data source
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if isFiltering() {
+            return filteredNotes.count
+        }
+        return NoteController.shared.notes.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "noteCell", for: indexPath)
+        let note = findCorrectNote(withIndex: indexPath)
+        cell.textLabel?.text = note.bodyText
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+        if isFiltering() {
+            if editingStyle == .delete {
+                let note = findCorrectNote(withIndex: indexPath)
+                NoteController.shared.deleteNote(note)
+                tableView.reloadData()
+                tableView.deleteRows(at: [indexPath], with: .fade)
```

Comment:
I'm not 100% sure because I don't do iOS anymore but wouldn't `reloadData()` update the table view to no longer have the rows you're about to delete in the next line? I don't recall the subtleties here.